packages/dashboard/src/app/views/myStories/filters/taxonomy/useTaxonomyFilters.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  useCallback,
+  useEffect,
+  useState,
+  useMemo,
+} from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import useApi from '../../../../api/useApi';
+
+export const getFilterPlaceholder = (filter) => {
+  return filter.labels?.allItems || filter.labels?.name || filter.restBase;
+};
+
+export const getFilterAriaLabel = (filter) => {
+  const ariaLable =
+    filter.labels?.singularName || filter.labels?.name || filter.restBase;
+  return `Filter stories by ${ariaLable}`;
+};
+
+export const getFilterSearchResultsLabel = (filter) => {
+  return (
+    filter.labels?.searchItems ||
+    (filter.labels?.name ? `Search ${filter.labels.name}` : 'Search Taxonomy')
+  );
+};
+
+/**
+ * Hook used for taxonomy filters logic
+ * Initilizes the taxonomy filters data
+ *
+ * @return {Array} [taxonomies, queryTaxonomyTerm] initial taxonomy filters data and a function to query the terms
+ */
+
+function useTaxonomyFilters() {
+  const [taxonomies, setTaxonomies] = useState([]);
+
+  const { getTaxonomies, getTaxonomyTerm } = useApi(
+    ({
+      actions: {
+        taxonomyApi: { getTaxonomies, getTaxonomyTerm },
+      },
+    }) => ({
+      getTaxonomies,
+      getTaxonomyTerm,
+    })
+  );
+
+  /**
+   * Query individual taxonomy data
+   * This is needed to initilize the primaryOptions and needed to search and set queriedOptions
+   *
+   * @param {Object} taxonomy object holding taxonomy data, including restBase and restPath used to fetch individual taxonomy terms
+   * @param {string|undefined} search string use to query taxonomy by name
+   * @return {Array} taxonomy terms
+   */
+  const queryTaxonomyTerm = useCallback(
+    async (taxonomy, search) => {
+      const { restBase, restPath } = taxonomy;
+      const terms = await getTaxonomyTerm(restPath, { search });
```

Comment:
```suggestion
      const terms = await getTaxonomyTerm(restPath, { search, per_page: -1 });
```

Let's get all terms here, no little to 100.